workbench-core/datasets-ui/src/components/CreateInternalDatasetForm.tsx

```diff
@@ -0,0 +1,282 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ProjectItem } from '@aws/workbench-core-accounts-ui/lib/esm/models/Project';
+import { validateField, FileUpload } from '@aws/workbench-core-swb-common-ui';
+import {
+  Box,
+  SpaceBetween,
+  Form,
+  Header,
+  ExpandableSection,
+  Input,
+  FormField,
+  Button,
+  Container,
+  Textarea,
+  Select,
+  TextContent,
+  Link
+} from '@cloudscape-design/components';
+import React, { useEffect, useState } from 'react';
+import { CreateDatasetForm, CreateDatasetFormValidation } from '../models/Dataset';
+import {
+  datasetDescriptionValidationRules,
+  datasetNameValidationRules,
+  datasetProjectIdValidationRules
+} from '../utils';
+
+interface NewDatasetFormProps {
+  onFormSubmit: (formData: CreateDatasetForm) => Promise<void>;
+  projects: ProjectItem[];
+  areProjectsLoading: boolean;
+}
+
+export const NewDatasetForm = ({
+  onFormSubmit,
+  projects,
+  areProjectsLoading
+}: NewDatasetFormProps): JSX.Element => {
+  const [isSubmitLoading, setIsSubmitLoading] = useState(false);
+  const [disableSubmit, setDisableSubmit] = useState(true);
+  const [error, setError] = useState('');
+  const [formData, setFormData] = useState<CreateDatasetForm>({});
+  const [formErrors, setFormErrors] = useState<CreateDatasetFormValidation>({});
```

Comment:
same as above for optionality.

Reply:
I'd keep empty object to avoid issues with NRE